### PR TITLE
[sonic_thermal]add abstract class for thermal plugin

### PR DESCRIPTION
**Pull request by stephenxs**
**What I did**
Add an abstract class for thermal plugin
**How I did it**
Add the following interfaces for thermal plugin
- get_num_thermals
- get_name
- get_temperature
- get_low_threshold
- get_low_critical_threshold
- get_high_threshold
- get_high_critical_threshold

### CODE REVIEW

File: sonic_thermal/thermal_base.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        """
+        return 0
+
+    def get_name(self, index):
+        """
+        Retrieves the human-readable name of a thermal sensor by 1-based index
+
+        Returns:
+        :param index: An integer, 1-based index of the thermal sensor of which to query status
+        :return: Float,
```
Comment (keboliu) [inline sonic_thermal/thermal_base.py]: should return a string of the name?

File: sonic_thermal/thermal_base.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+#
+# thermal_base.py
+#
+# Base class for implementing platform-specific
+#  Thermal control functionality for SONiC
```
Comment (keboliu) [inline sonic_thermal/thermal_base.py]: rephrase?

File: sonic_thermal/thermal_base.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        :return: Float,
+            A float number of current temperature in Celsius up to nearest thousandth
+            of one degree Celsius, e.g. 30.125 
+        """
+        return 0.0
+
+    def get_low_threshold(self, index):
+        """
+        Retrieves the low threshold temperature of thermal sensor by 1-based index
+        Steps should be taken if the temperature becomes lower than the low threshold.
```
Comment (keboliu) [inline sonic_thermal/thermal_base.py]: steps --> action?  same comments to below.

**Review by keboliu — CHANGES_REQUESTED** [review]
as comments

File: sonic_thermal/thermal_base.py (hunk 1 above)
Comment (stephenxs) [reply sonic_thermal/thermal_base.py]: fixed

File: sonic_thermal/thermal_base.py (hunk 2 above)
Comment (stephenxs) [reply sonic_thermal/thermal_base.py]: fixed

File: sonic_thermal/thermal_base.py (hunk 3 above)
Comment (stephenxs) [reply sonic_thermal/thermal_base.py]: fixed

**Review by keboliu — APPROVED** [review]

**Review by jleveque — APPROVED** [review]

**Comment by stephenxs** [comment]
Community PR [#75](https://github.com/Azure/sonic-platform-common/pull/75) opened.

**Pull request closed without merging**